### PR TITLE
Dev initial refactoring

### PR DESCRIPTION
**Pull request by taraspiotr**

### CODE REVIEW

File: src/models.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
 import numpy as np
 import torch.optim as optim
 
-from steps.pytorch.architectures.unet import UNet, UNetMultitask
-from steps.pytorch.callbacks import CallbackList, TrainingMonitor, ValidationMonitor, ModelCheckpoint, \
+from .steppy.pytorch.architectures.unet import UNet, UNetMultitask
```
Comment (jakubczakon) [inline src/models.py]: @taraspiotr I think we will go with Resnet101/152 so not sure how usefull are those 2

File: src/pipelines.py [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-from loaders import MetadataImageSegmentationLoader, MetadataImageSegmentationMultitaskLoader, \
+from .steppy.base import Step, Dummy
+from .steppy.preprocessing.misc import XYSplit, ImageReader
+
+from .loaders import MetadataImageSegmentationLoader, MetadataImageSegmentationMultitaskLoader, \
     ImageSegmentationMultitaskLoader, ImageSegmentationLoader
-from models import PyTorchUNet, PyTorchUNetMultitask
-from postprocessing import Resizer, Thresholder, NucleiLabeler, Dropper, \
+from .models import PyTorchUNet, PyTorchUNetMultitask
+from .postprocessing import Resizer, Thresholder, NucleiLabeler, Dropper, \
```
Comment (jakubczakon) [inline src/pipelines.py]: @taraspiotr I am pretty sure that WateshedContour is the only postprocessing needed (maybe BinaryFillHoles but nothing else)

File: src/steppy/base.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+        return NotImplementedError
+
+    def fit_transform(self, *args, **kwargs):
+        self.fit(*args, **kwargs)
+        return self.transform(*args, **kwargs)
+
+    def load(self, filepath):
+        return self
+
+    def save(self, filepath):
```
Comment (jakubczakon) [inline src/steppy/base.py]: @taraspiotr I think steppy transformers have .persist()

File: src/steppy/base.py [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+        self.fit(*args, **kwargs)
+        return self.transform(*args, **kwargs)
+
+
+class Dummy(BaseTransformer):
+    def transform(self, **kwargs):
+        return kwargs
+
+
+def to_tuple_inputs(inputs):
```
Comment (jakubczakon) [inline src/steppy/base.py]: @taraspiotr all of those adapter functions are not existent in steppy anymore. You simply create a step for that

File: src/steppy/keras/contrib.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+from os.path import dirname
+
+sys.path.append(dirname(dirname(__file__)))
+from keras import initializers
+from keras.engine import InputSpec, Layer
+from keras import backend as K
+import tensorflow as tf
+
+
+class AttentionWeightedAverage(Layer):
```
Comment (jakubczakon) [inline src/steppy/keras/contrib.py]: @taraspiotr I'd drop the entire steps/keras from this project

File: src/steppy/keras/embeddings.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
+import numpy as np
```
Comment (jakubczakon) [inline src/steppy/keras/embeddings.py]: @taraspiotr this would go as well

File: src/steppy/keras/loaders.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
+from keras.preprocessing import text, sequence
```
Comment (jakubczakon) [inline src/steppy/keras/loaders.py]: @taraspiotr this would go as well

File: src/steppy/keras/models.py [hunk 8]
```diff
@@ -0,0 +1,213 @@
+import shutil
```
Comment (jakubczakon) [inline src/steppy/keras/models.py]: @taraspiotr this would go as well

File: src/steppy/preprocessing/text.py [hunk 9]
```diff
@@ -0,0 +1,242 @@
+import json
```
Comment (jakubczakon) [inline src/steppy/preprocessing/text.py]: @taraspiotr this would go as well

File: src/steppy/resources/apostrophes.json [hunk 10]
```diff
@@ -0,0 +1,58 @@
+{
```
Comment (jakubczakon) [inline src/steppy/resources/apostrophes.json]: @taraspiotr this would go as well

File: src/steppy/sklearn/models.py [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+    @property
+    def estimator(self):
+        return CatBoostClassifier
+
+
+class XGBoostClassifierMultilabel(MultilabelEstimator):
+    @property
+    def estimator(self):
+        return XGBClassifier
+
```
Comment (jakubczakon) [inline src/steppy/sklearn/models.py]: @taraspiotr this would go as well

**Comment by taraspiotr** [comment]
@jakubczakon sure, I am going to clean all of that as well, in this PR I wanted only to include steppy and make this branch "runable", I will drop this PR and pull another one with solution 0 ready.

**Pull request closed without merging**